### PR TITLE
sd-bus: drop version 2 format support

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/basic/macro.h [hunk 1]
```diff
@@ -112,8 +112,23 @@
...
 /* Rounds up */
 
-#define ALIGN4(l) (((l) + 3) & ~3)
-#define ALIGN8(l) (((l) + 7) & ~7)
+#define _ALIGN(l, ali)                                          \
+        ({                                                      \
+                typeof(l) _l = -1;                              \
+                                                                \
+                assert_cc(ali == 4 || ali == 8);                \
+                assert(_l > 0); /* check unsigned */            \
```
Comment (medhefgo) [inline src/basic/macro.h]: `assert_cc(IS_UNSIGNED_INTEGER_TYPE(typeof(l));`

Or is this supposed to allow signed types?

File: src/test/test-macro.c [hunk 2]
```diff
@@ -290,6 +290,89 @@ TEST(foreach_pointer) {
...
+        assert_se(ALIGN4((size_t) 4) == 4);
+        assert_se(ALIGN4((size_t) 5) == 8);
+        assert_se(ALIGN4((size_t) SIZE_MAX-4) == SIZE_MAX-3);
+        assert_se(ALIGN4((size_t) SIZE_MAX-3) == SIZE_MAX-3);
+        assert_se(ALIGN4((size_t) SIZE_MAX-2) == SIZE_MAX);
+        assert_se(ALIGN4((size_t) SIZE_MAX-1) == SIZE_MAX);
+        assert_se(ALIGN4((size_t) SIZE_MAX) == SIZE_MAX);
+
+        assert_se(ALIGN4((unsigned) 0) == 0);
+        assert_se(ALIGN4((unsigned) 1) == 4);
```
Comment (medhefgo) [inline src/test/test-macro.c]: `1U` etc?

File: src/test/test-macro.c [hunk 3]
```diff
@@ -290,6 +290,89 @@ TEST(foreach_pointer) {
...
+        assert_se(ALIGN4((unsigned) 3) == 4);
+        assert_se(ALIGN4((unsigned) 4) == 4);
+        assert_se(ALIGN4((unsigned) 5) == 8);
+        assert_se(ALIGN4((unsigned) UINT_MAX-4) == UINT_MAX-3);
+        assert_se(ALIGN4((unsigned) UINT_MAX-3) == UINT_MAX-3);
+        assert_se(ALIGN4((unsigned) UINT_MAX-2) == UINT_MAX);
+        assert_se(ALIGN4((unsigned) UINT_MAX-1) == UINT_MAX);
+        assert_se(ALIGN4((unsigned) UINT_MAX) == UINT_MAX);
+
+        assert_se(ALIGN4((uint64_t) 0) == 0);
```
Comment (medhefgo) [inline src/test/test-macro.c]: `UINT64_C()`?

File: src/basic/macro.h (hunk 1 above)
Comment (medhefgo) [reply src/basic/macro.h]: Also, this seems to do the exact same thing as ALIGN_TO. Maybe just call that?

Comment (yuwata) [reply src/basic/macro.h]: Yeah, that sounds better.

**Review by medhefgo — APPROVED** [review]

**Pull request merged**